udata_hydra/datalake_service.py

```diff
@@ -132,3 +150,31 @@ async def process_resource(url: str, dataset_id: str, resource_id: str, response
     finally:
         if tmp_file:
             os.remove(tmp_file.name)
+        return {
+            "checksum": sha1,
+            "error": error,
+            "filesize": filesize,
+            "mime_type": mime_type
+        }
+
+
+async def has_checksum_been_modified(resource_id, new_checksum):
+    q = """
+    SELECT
+        checksum
+    FROM checks
+    WHERE resource_id = $1 and checksum IS NOT NULL
+    ORDER BY created_at DESC
+    LIMIT 1
+    """
+    pool = await context.pool()
+    async with pool.acquire() as connection:
+        data = await connection.fetch(q, resource_id)
+        if data:
+            if data[0]['checksum'] != new_checksum:
+                return True
+            else:
+                return False
+        else:
+            # First check, thus this is a new checksum
```

Comment:
Unsure about that 🤔 It doesn't mean the file has changed. We can iterate on `Last-Modified` header later if checksum is not conclusive.

Reply:
Indeed, it would be misleading to expose that the file has changed in this case.
I've modified the behaviour in https://github.com/etalab/udata-hydra/pull/32/commits/0fe46fc0352b7e38b15f4361cbac54cedde83696.